jparest/src/main/java/uk/gov/homeoffice/digital/sas/jparest/service/ResourceApiService.java

```diff
@@ -147,23 +75,9 @@ public void deleteResource(UUID tenantId, UUID id) {
         this.transactionManager.getTransaction(transactionDefinition);
 
     try {
-      var builder = this.entityManager.getCriteriaBuilder();
-      CriteriaDelete<T> query = builder.createCriteriaDelete(this.entityUtils.getEntityType());
-      Root<T> root = query.from(this.entityUtils.getEntityType());
-
-      var tenantPredicate = builder.equal(root.get(TENANT_ID.getParamName()), tenantId);
-      var idPredicate = builder.equal(root.get(ID.getParamName()), id);
-      query.where(builder.and(tenantPredicate, idPredicate));
-
-      var updatedItems = this.entityManager.createQuery(query).executeUpdate();
-      if (updatedItems == 0) {
-        throw new EmptyResultDataAccessException(1);
-      } else if (updatedItems > 1) {
-        throw new UnexpectedQueryResultException(id);
-      }
-
+      repository.deleteByTenantIdAndId(tenantId, id);
       transactionManager.commit(transactionStatus);
-    } catch (EmptyResultDataAccessException ex) {
+    } catch (NoSuchElementException ex) {
       transactionManager.rollback(transactionStatus);
       throw new ResourceNotFoundException(id);
```

Comment:
This won't be needed if enclosing method is annotated with @Transactional

Reply:
part of the next subtask eahw-2545